crates/dbsp/src/trace/layers/column_layer/mod.rs

```diff
@@ -202,8 +215,8 @@ impl<K, R> ColumnLayer<K, R> {
         let mut shifter = BackshiftOnDrop {
             keys: &mut self.keys,
             diffs: &mut self.diffs,
-            processed_len: 0,
-            deleted_cnt: 0,
+            processed_len: self.lower_bound,
+            deleted_cnt: self.lower_bound,
```

Comment:
I'm not sure this is correct, this will leak all items at `0..lower_bound`

Reply:
Thanks for catching this.  Does the new version look better?